pinot-core/src/test/java/com/linkedin/pinot/core/startree/hll/SegmentWithHllIndexCreateHelper.java

```diff
@@ -15,144 +15,152 @@
-import org.apache.avro.generic.GenericDatumReader;
-import org.apache.avro.generic.GenericRecord;
-import org.apache.commons.io.FileUtils;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
 import java.io.File;
-import java.io.FileInputStream;
 import java.io.IOException;
 import java.nio.file.Files;
 import java.util.concurrent.TimeUnit;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class SegmentWithHllIndexCreateHelper {
-    private static final Logger LOGGER = LoggerFactory.getLogger(SegmentWithHllIndexCreateHelper.class);
-
-    private File INDEX_DIR;
-    private File inputAvro;
-    private String timeColumnName;
-    private TimeUnit timeUnit;
-    private Schema schema;
-    private static final String hllDeriveColumnSuffix = HllConstants.DEFAULT_HLL_DERIVE_COLUMN_SUFFIX;
-
-    public SegmentWithHllIndexCreateHelper(String tempDirName, String avroDataPath,
-                                           String timeColumnName, TimeUnit timeUnit) throws IOException {
-        INDEX_DIR = Files.createTempDirectory(SegmentWithHllIndexCreateHelper.class.getName() + "_" + tempDirName).toFile();
-        LOGGER.info("INDEX_DIR: {}", INDEX_DIR.getAbsolutePath());
-        String filePath = TestUtils.getFileFromResourceUrl(
-                SegmentV1V2ToV3FormatConverter.class.getClassLoader().getResource(avroDataPath));
-        inputAvro = new File(filePath);
-        LOGGER.info("Input Avro: {}", inputAvro.getAbsolutePath());
-        this.timeColumnName = timeColumnName;
-        this.timeUnit = timeUnit;
+  private static final Logger LOGGER = LoggerFactory.getLogger(SegmentWithHllIndexCreateHelper.class);
+  private static final String hllDeriveColumnSuffix = HllConstants.DEFAULT_HLL_DERIVE_COLUMN_SUFFIX;
+
+  private final String tableName;
+  private final File INDEX_DIR;
+  private final File inputAvro;
+  private final String timeColumnName;
+  private final TimeUnit timeUnit;
+  private final String segmentName = "starTreeSegment";
+  private Schema schema;
+
+  public SegmentWithHllIndexCreateHelper(String tableName, String avroDataPath,
+      String timeColumnName, TimeUnit timeUnit) throws IOException {
+    INDEX_DIR = Files.createTempDirectory(SegmentWithHllIndexCreateHelper.class.getName() + "_" + tableName).toFile();
+    LOGGER.info("INDEX_DIR: {}", INDEX_DIR.getAbsolutePath());
+    String filePath = TestUtils.getFileFromResourceUrl(getClass().getClassLoader().getResource(avroDataPath));
+    inputAvro = new File(filePath);
+    LOGGER.info("Input Avro: {}", inputAvro.getAbsolutePath());
+    this.timeColumnName = timeColumnName;
+    this.timeUnit = timeUnit;
+    this.tableName = tableName;
+  }
+
+  /**
+   * must call this to clean up
+   */
+  public void cleanTempDir() {
+    if (INDEX_DIR != null) {
+      FileUtils.deleteQuietly(INDEX_DIR);
+    }
+  }
+
+  private static void printSchema(Schema schema) {
```

Comment:
Still I think these are info level information. e.g. directory, column names and so on. Not the debug level information, e.g. intermediate results and so on.